File: backend/src/main/java/com/example/budgeter/entity/Budget.java
package com.example.budgeter.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
@Entity
@Table(name = "budgets")
public class Budget {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id", nullable = false)
  private Integer id;

  @Size(max = 40)
  @NotNull
  @Column(name = "name", nullable = false, length = 40)
  private String name;

  @NotNull
  @Column(name = "period_start", nullable = false)
  private LocalDate periodStart;

  @NotNull
  @Column(name = "period_end", nullable = false)
  private LocalDate periodEnd;

  @Column(name = "total_limit", precision = 12, scale = 2)
  private BigDecimal totalLimit;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "budget_category_id")
  private BudgetCategory budgetCategory;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "id_user")
  private User idUser;


}

File: frontend/src/lib/Budget.ts
import { BigDecimal } from "./additionalTypes/BigDecimal";
import { Transaction } from "./Transaction";

export interface BudgetDto {
  id?: number | null;
  name?: string | null;
  periodStart?: string | null;
  periodEnd?: string | null;
  totalLimit?: BigDecimal | null;
  transactions?: Transaction[] | null;
}

export interface BudgetRequest {
  name?: string | null;
  periodStart?: string | null;
  periodEnd?: string | null;
  totalLimit?: BigDecimal | null;
}

export interface BudgetUpdateRequest extends BudgetRequest {
  id?: number | null;
}